apimodels/task_log.go

```diff
@@ -35,9 +51,107 @@ type TaskLog struct {
 	Messages     []LogMessage `json:"m"`
 }
 
-// BuildloggerLogLine represents a cedar buildlogger log line, with the
-// severity extracted.
-type BuildloggerLogLine struct {
-	Message  string `bson:"m" json:"m"`
-	Severity string `bson:"s" json:"s"`
+func GetSeverityMapping(s int) string {
+	switch {
+	case s >= int(level.Error):
+		return LogErrorPrefix
+	case s >= int(level.Warning):
+		return LogWarnPrefix
+	case s >= int(level.Info):
+		return LogInfoPrefix
+	case s < int(level.Info):
+		return LogDebugPrefix
+	default:
+		return LogInfoPrefix
+	}
+}
+
+func GetBuildloggerLogs(ctx context.Context, buildloggerBaseURL, taskId, logType string, tail, execution int) (io.ReadCloser, error) {
+	usr := gimlet.GetUser(ctx)
+	opts := fetcher.GetOptions{
+		BaseURL:       fmt.Sprintf("https://%s", buildloggerBaseURL),
+		UserKey:       usr.GetAPIKey(),
+		UserName:      usr.Username(),
+		TaskID:        taskId,
+		Execution:     execution,
+		PrintTime:     true,
+		PrintPriority: true,
+		Tail:          tail,
+	}
+	switch logType {
+	case TaskLogPrefix:
+		opts.ProcessName = evergreen.LogTypeTask
+	case SystemLogPrefix:
+		opts.ProcessName = evergreen.LogTypeSystem
+	case AgentLogPrefix:
+		opts.ProcessName = evergreen.LogTypeAgent
+	}
+
+	logReader, err := fetcher.Logs(ctx, opts)
+	return logReader, errors.Wrapf(err, "failed to get logs for '%s' from buildlogger, using evergreen logger", taskId)
+}
+
+func ReadBuildloggerToChan(ctx context.Context, taskID string, r io.ReadCloser, lines chan<- LogMessage) {
+	var (
+		line string
+		err  error
+	)
+
+	defer close(lines)
+	if r == nil {
+		return
+	}
+
+	reader := bufio.NewReader(r)
+	for err == nil {
```

Comment:
@john-m-liu do you think there is a better way to handle this error?

Reply:
This is written in an unusual way because this check is logically `while (we have not reached the end of the file)` so discarding the error is the right thing to do here. It's also unchanged code so I don't think it's worth worrying about in this PR